iotdb-core/node-commons/src/test/java/org/apache/iotdb/commons/utils/NodeUrlUtilsTest.java

```diff
@@ -59,4 +59,37 @@ public void parseAndConvertTConfigNodeUrlsTest() throws BadNodeUrlException {
     Assert.assertEquals(configNodeUrls, NodeUrlUtils.convertTConfigNodeUrls(configNodeLocations));
     Assert.assertEquals(configNodeLocations, NodeUrlUtils.parseTConfigNodeUrls(configNodeUrls));
   }
+
+  @Test
+  public void parseAndConvertTEndPointUrlsIPV4AndIPV6Test() throws BadNodeUrlException {
+    final List<TEndPoint> endPoints =
+        Arrays.asList(
+            new TEndPoint("AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:2345", 6667),
+            new TEndPoint("0:0:0:0:0:FFFF:129.144.52.38", 6668),
+            new TEndPoint("::13.1.68.3", 6669));
+    final String endPointUrls =
+        "AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:2345:6667,[0:0:0:0:0:FFFF:129.144.52.38]:6668,[::13.1.68.3]:6669";
+    Assert.assertEquals(endPoints, NodeUrlUtils.parseTEndPointUrls(endPointUrls));
+  }
+
+  @Test
+  public void parseAndConvertTConfigNodeUrlsIPV4AndIPV6Test() throws BadNodeUrlException {
+    final List<TConfigNodeLocation> configNodeLocations =
+        Arrays.asList(
+            new TConfigNodeLocation(
+                0,
+                new TEndPoint("AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD", 22277),
+                new TEndPoint("AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD", 22278)),
+            new TConfigNodeLocation(
+                1,
+                new TEndPoint("AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD", 22279),
+                new TEndPoint("AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD", 22280)),
+            new TConfigNodeLocation(
+                2,
+                new TEndPoint("AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD", 22281),
+                new TEndPoint("AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD", 22282)));
+    final String configNodeUrls =
+        "0,AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD:22277,[AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD]:22278;1,AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD:22279,AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD:22280;2,AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD:22281,AD80:E32B:CR25:B3WE:DG4G:DWTF:CGDE:ABFD:22282";
```

Comment:
[], and A-F

Reply:
If IoTDB supports the no '[]' version, I think it is still OK without it.

Reply:
> If IoTDB supports the no '[]' version, I think it is still OK without it.

Whether to support it needs discussion?

Reply:
Take all the circumstances into account

Reply:
Ok, I agree that though IPv6 protocal may not contains this, supporting this maybe friendlier for users.